src/lib/snark_work_lib/selector.ml

```diff
@@ -0,0 +1,82 @@
+open Core_kernel
+
+module Single = struct
+  module Spec = struct
+    [%%versioned
+    module Stable = struct
```

Comment:
Do we need to define it as stable type?

Would simpler definition of like the following work?

```ocaml
type t =
  ( Transaction_witness.t
  , Ledger_proof.Cached.t )
  Snark_work_lib.Work.Single.Spec.t
```

Reply:
The reason every type in `Snark_work_lib.Selector` and `Snark_work_lib.Partitioned` are stable type, is that these types would be passed around between RPC endpoints. I'll update the in-code documentation.